src/UpdatesLog.php

```diff
@@ -108,7 +109,15 @@ public function run(): void {
       $this->state->set(self::STATUSES_STATE, $new_statuses);
     }
     if (getenv('UPDATES_LOG_TEST') || ($now >= $this->getLastRanStatistics() + (60 * 60 * 24))) {
-      $statistics = $this->generateStatistics($statuses);
+      $version = $this->getVersion();
+      $site = $this->getSite();
+      $env = $this->getEnv();
```

Comment:
```
$version = $this->getVersion();
$site = $this->getSite();
$env = $this->getEnv();
```
I would move all of this into "generateStatistics" 
IMO it is "ugly" to use function with 4 arguments when 3 of them are only used inside the function

Reply:
The proposed approach would have drawbacks:
* The `generateStatistics()` would not be atomic anymore
* The `generateStatistics()` would be harder to test
* The `generateStatistics()` test would need to be moved into Kernel Tests because of `getVersion()`.
* Violates Business-Logic and Storage separation